src/style/style.test.ts

```
@@ -172,6 +172,32 @@ describe('Style#loadURL', () => {
         style._remove();
         expect((server.lastRequest as any).aborted).toBe(true);
     });
+
+    test('does not fire an error if removed', async () => {
+        const style = new Style(getStubMap());
+        const spy = jest.fn();
+
+        style.on('error', spy);
```

you can use `once` and expect that the promise will not be rejected, instead of `sleep(0)`

---

I'm not sure how to do that. Since we expect the `style.once('error')` promise to neither resolves nor rejects (because `this.fire` should never be called), we can't await it. And since the abort happens asynchronously, we can't test the spy function immediately after removing the style. Ideally, I would wait for the abort error to be thrown, but I couldn't find a way to do that with the nise fake requests.

Any advice would be welcomed, as I'm not very familiar with these types of tests.

---

Ahh, I see, I misread the code, can you change it to `not.toHaveBeenCalled` instead of times 0?

---

Also, I think the following example raises an abort signal: 
https://github.com/maplibre/maplibre-gl-js/blob/6a996a11958a85b6a2192b4f9957b5b27348b495/src/source/vector_tile_worker_source.test.ts#L35

I'm not 100% sure as I'm not sure this is actually documented...

---

Of course, I'll change it to `not.toHaveBeenCalled`.

I don't fully understand this test, but I think one key difference is that `source.loadTile` ans `source.abortTile` both return promises that can be awaited, while `style.loadURL` and `style._remove` don't. So I'm not sure the same method can be applied here.

---

I think calling this abort url will cause an abort signal, which you wrote you needed, not that important though.

---

After doing some test on `vector_tile_worker_source.test.ts`, it seems replacing the URL by any string doesn't change the test result. So I think it's just an arbitrary fake URL.

I managed to make it work with this however : 
```js
style.on('error', spy);
style.loadURL('style.json');
const promise = new Promise((_, reject) => {
    (server.lastRequest as any).addEventListener('abort', reject);
});
style._remove();
await promise.catch(() => {});

expect(spy).not.toHaveBeenCalled();
```
It might not be the prettiest code, but just awaiting the promise doesn't work, a `then` or `catch` is needed. It can also be done like that : 
```js
const promise = new Promise((resolve) => {
    (server.lastRequest as any).addEventListener('abort', resolve);
});
...
await promise.then(() => {});
```
IMO, I find the `reject/catch` to be cleaner, because it mirrors more accurately the behaviour of the request, and adding a catch also serves the purpose of ignoring the error.

I will commit with these changes as I won't be able to work on it for a few days. I'll come back to it later if it needs other changes.